src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/PairedEndAndSplitReadEvidenceCollection.java

```diff
@@ -0,0 +1,434 @@
+            } else {
+                srWriter = new OutputStreamWriter(new FileOutputStream(new File(srFile)));
+            }
+        } catch (FileNotFoundException e) {
+            throw new UserException("Could not open " + srFile);
+        }
+
+        splitPosBuffer = new PriorityQueue<>(new SplitPosComparator());
+
+        discordantPairs = new ArrayList<>();
+
+        sequenceDictionary = getBestAvailableSequenceDictionary();
+    }
+
+    public boolean isExcluded(final GATKRead read) {
+        return read.isUnmapped() || read.mateIsUnmapped() || read.isSecondaryAlignment() || read.isDuplicate() || read.isSupplementaryAlignment();
+    }
+
+
+    @Override
+    public void apply(final GATKRead read, final ReferenceContext referenceContext, final FeatureContext featureContext) {
+        if (isExcluded(read)) {
+            return;
+        }
+
+        if (isSoftClipped(read)) {
+            countSplitRead(read, splitPosBuffer, srWriter);
+        }
+
+        if (! read.isProperlyPaired()) {
+            reportDiscordantReadPair(read);
+        }
+    }
+
+    private void reportDiscordantReadPair(final GATKRead read) {
+        if (read.getStart() != currentDiscordantPosition) {
+            flushDiscordantReadPairs();
+            currentDiscordantPosition = read.getStart();
+            observedDiscordantNames.clear();
+        }
+
+        final DiscordantRead reportableDiscordantReadPair = getReportableDiscordantReadPair(read, observedDiscordantNames,
+                sequenceDictionary);
+        if (reportableDiscordantReadPair != null) {
+            discordantPairs.add(reportableDiscordantReadPair);
+        }
+    }
+
+    @VisibleForTesting
+    public DiscordantRead getReportableDiscordantReadPair(final GATKRead read, final Set<String> observedDiscordantNamesAtThisLocus,
+                                                          final SAMSequenceDictionary samSequenceDictionary) {
+        final int readSeqId = samSequenceDictionary.getSequenceIndex(read.getContig());
+        final int mateSeqId = samSequenceDictionary.getSequenceIndex(read.getMateContig());
+        if (readSeqId < mateSeqId) {
+            return new DiscordantRead(read);
+        } else if (readSeqId == mateSeqId) {
+            if (read.getStart() < read.getMateStart()) {
+                return new DiscordantRead(read);
+            } else if (read.getStart() == read.getMateStart()) {
+                final boolean seenBefore = observedDiscordantNamesAtThisLocus.remove(read.getName());
+                if (! seenBefore) {
+                    final DiscordantRead discordantRead = new DiscordantRead(read);
+                    observedDiscordantNamesAtThisLocus.add(read.getName());
+                    return discordantRead;
+                }
+            }
+        }
+        return null;
+    }
+
+    private void flushDiscordantReadPairs() {
+        final Comparator<DiscordantRead> discReadComparator =
```

Comment:
Can you make a class out of this? Could be useful in the future

Reply:
Wrapped this up in a static inner class for now.